core/src/main/java/org/infinispan/commands/write/ApplyDeltaCommand.java

```diff
@@ -25,7 +25,7 @@ public ApplyDeltaCommand() {
    }
 
    public ApplyDeltaCommand(Object deltaAwareValueKey, Delta delta, Collection<Object> keys, CommandInvocationId commandInvocationId) {
-      super(deltaAwareValueKey, 0, commandInvocationId);
+      super(deltaAwareValueKey, -1, 0, commandInvocationId);
```

Comment:
Will `segment = -1` not always throw an `IllegalArgumentException` in `AbstractDataCommand`?

Reply:
yes, tbh this code is all dead anyways as we don't use it anymore. I wasn't that worried about it.

Reply:
`AdvancedCache.applyDelta` doesn't even use the `ApplyDeltaCommand` :D 

https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/cache/impl/CacheImpl.java#L1018

Reply:
And the constructor is never invoked and is in a private package. I can add a segment argument if you guys wanted, since it doesn't really matter.

Reply:
Ah yes, I guess it's still around only because it's in the signature of `Visitor.visitApplyDeltaCommand()` and the custom interceptors. Since all the methods already throw `UnsupportedOperationException`, I think we should also throw USO in the constructor, and explain in the javadoc why the class is still around, because to me "deprecated" means it's not recommended, but it still works if you need it.